### ISSUE TITLE
`codegen-backends = ["rustc_codegen_cranelift"]` should give a better error message

### ISSUE THREAD

**Issue by jyn514** [rust-lang/rust#109315]
I tried this code:

```rust
echo 'codegen-backends = ["rustc_codegen_cranelift"]` >> config.toml
x test ui
```

I expected to see this happen: An error saying that `rustc_codegen_cranelift` is invalid and should be `cranelift` instead.

Instead, this happened:
```
Building stage0 codegen backend rustc_codegen_cranelift
error: manifest path `/home/jyn/src/rust2/compiler/rustc_codegen_rustc_codegen_cranelift/Cargo.toml` does not exist
```

### Meta

HEAD is branched from https://github.com/rust-lang/rust/commit/13afbdaa0655dda23d7129e59ac48f1ec88b2084
